Site/SiteAnalyticsModule.php

```diff
@@ -213,7 +242,126 @@ public function getGoogleAnalyticsTrackerInlineJavascript()
 
 			$javascript = sprintf(
 				$javascript,
-				$this->getTrackingCodeSource()
+				$this->getGoogleAnalyticsTrackingCodeSource()
+			);
+		}
+
+		return $javascript;
+	}
+
+	// }}}
+	// {{{ public function hasGoogleAnalytics()
+
+	public function hasFacebookPixel()
+	{
+		return (
+			$this->facebook_pixel_id != '' &&
+			$this->analytics_opt_out === false
+		);
+	}
+
+	// }}}
+	// {{{ public function pushFacebookPixelCommand()
+
+	public function pushFacebookPixelCommand($command)
+	{
+		$this->facebook_pixel_commands = array_merge(
+			$this->facebook_pixel_commands,
+			$command
+		);
+	}
+
+	// }}}
+	// {{{ public function prependFacebookPixelCommand()
+
+	public function prependFacebookPixelCommand($command)
+	{
+		array_unshift($this->facebook_pixel_commands, $command);
+	}
+
+	// }}}
+	// {{{ public function getFacebookPixelImage()
+
+	public function getFacebookPixelImage()
+	{
+		return sprintf(
+			'<noscript><img height="1" width="1" style="display:none"
+			src="https://www.facebook.com/tr?id=%s&ev=PageView&noscript=1"
+			/></noscript>',
```

Comment:
Spanning the newlines inside the tag string causes them to be included in the output. It would be cleaner to use string concatenation here.